python/pyspark/pandas/data_type_ops/datetime_ops.py

```diff
@@ -135,9 +136,7 @@ def astype(self, index_ops: IndexOpsLike, dtype: Union[str, type, Dtype]) -> Ind
         elif isinstance(spark_type, StringType):
             if isinstance(dtype, extension_dtypes):
                 # seems like a pandas' bug?
-                scol = F.when(index_ops.spark.column.isNull(), str(pd.NaT)).otherwise(
-                    index_ops.spark.column.cast(spark_type)
-                )
+                return _as_string_type(index_ops, dtype)
```

Comment:
Good point, `extension_dtypes` check has been done in `_as_string_type`.

Reply:
It would be nice to say that `null_str` is for non-extension dtypes only in the docstring of `_as_string_type`. I may also file a follow-up PR if needed.

Reply:
Sounds good!

Reply:
> It would be nice to say that `null_str` is for non-extension dtypes only in the docstring of `_as_string_type`. I may also file a follow-up PR if needed.

I added !

Reply:
Thank you!